src/data/nat/factorial.lean

```diff
@@ -96,4 +96,22 @@ lemma self_le_factorial : ∀ n : ℕ, n ≤ n!
 | 0 := zero_le_one
 | (k+1) := le_mul_of_one_le_right k.zero_lt_succ.le (nat.one_le_of_lt $ nat.factorial_pos _)
 
+lemma lt_factorial_self {n : ℕ} (hi : 3 ≤ n) : n < n! :=
+begin
+  rw [← succ_pred_eq_of_pos (lt_of_lt_of_le (zero_lt_two.trans (lt.base 2)) hi), factorial_succ],
```

Comment:
You might be able to use dot notation a bit more here using the alias [`has_lt.lt.trans_le`](https://leanprover-community.github.io/mathlib_docs/algebra/order.html#has_lt.lt.trans_le).

Reply:
Thanks!  I did not know about this abbreviation!

Reply:
Yep, it's great! Looks like there's one more `lt_of_lt_of_le` on line 102; does dot notation work there?

Reply:
I think that I did it in a second push: it should be there!